libtalek/topic_test.go

```diff
@@ -60,6 +62,22 @@ func TestSerializeRestore(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Unable to restore topic: %v\n", err)
 	}
+
+	// test text encoder
+	txt, err := topic.MarshalText()
+	if err != nil {
+		t.Fatalf("Error serializing: %v\n", err)
+	}
+	fmt.Printf("Serialized topic looks like %s\n", txt)
+
+	clone = Topic{}
+	err = clone.UnmarshalText(txt)
+	if err != nil {
+		t.Fatalf("Could not deserialize: %v\n", err)
+	}
+	if !bytes.Equal(topic.SigningPrivateKey[:], clone.SigningPrivateKey[:]) || !bytes.Equal(topic.Handle.SharedSecret[:], clone.Handle.SharedSecret[:]) {
```

Comment:
Again, we're missing checks on some fields here. It seems like the right thing to do is for the topic and handle to both have their own Equal() method